src/ln/functional_tests.rs

```diff
@@ -1653,6 +1656,11 @@ fn channel_reserve_in_flight_removes() {
 fn channel_monitor_network_test() {
 	// Simple test which builds a network of ChannelManagers, connects them to each other, and
 	// tests that ChannelMonitor is able to recover from various states.
+	let mut block_1;
```

Comment:
nit: why `let` these in a block at the top? They aren't referenced after connecting, so just let them where they're assigned to?

Reply:
since adding the lifetime to the `Node` struct it seems to be enforcing the requirement that all of the data it borrows outlives it, meaning `block`s need to be defined before `node`s (because `values in a scope are dropped in the opposite order they are created`). 

It does add quite a bit to the diff so open to ways of subverting this..

Reply:
Ahhhh! This is an indication that something is pretty broken - you're defining lifetimes on the traits (the ChainListener trait) and then requiring method parameters live that long. ie the way the lifetimes are defined, any time we pass a block into filter_block/block_connected, the compiler will prove for us that the Block is around for at least as long as the ChainListener itself! It seems to work fine if I move the lifetimes to the function itself, ie
`fn filter_block<'a>(&self, block: &'a Block) -> (Vec<&'a Transaction>, Vec<u32>)`